src/queueloader.cpp

```diff
@@ -243,7 +243,7 @@ void QueueLoader::delete_played_files(const CategorizedDownloads& downloads)
 const
 {
 	for (const auto& dl : downloads.to_delete) {
-		const Filepath filename = dl.filename();
+		const Filepath filename = dl.filename().clone();
```

Comment:
Why do we need `clone()` here?

Reply:
Without `clone()` we got this. I'm not going into `Filepath` class yet. I tried create copy ctor/optor, but run into problems (compile-errors), later I will try to solve it.
```bash
src/queueloader.cpp:246:41: error: use of deleted function ‘newsboat::Filepath::Filepath(const newsboat::Filepath&)’
  246 |   const Filepath filename = dl.filename();
      |                                         ^
In file included from include/download.h:7,
                 from include/queueloader.h:10,
                 from src/queueloader.cpp:1:
include/filepath.h:89:2: note: declared here
   89 |  Filepath(const Filepath&) = delete;
      |  ^~~~~~~~
```

Reply:
I think this can be fixed by changing the signature of `Download::filename()` from:

```c++
const newsboat::Filename filename() const;
```

to 


```c++
newsboat::Filename filename() const;
```

That way C++ can just move the result.

Ditto for `dirname()`.

Reply:
> Ditto for `dirname()`.

You probably meant `basename()`

https://github.com/newsboat/newsboat/pull/2561/commits/9e0aea3c2bc826e5e807ce1656d12b305dfd0004